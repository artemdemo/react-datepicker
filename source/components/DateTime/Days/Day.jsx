import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { propIsMoment } from '../propTypes';

import './Day.less';

class Day extends React.Component {
    clickHandler(validDate) {
        const { date, onClick } = this.props;
        if (validDate) {
            onClick && onClick(date);
        }
    }

    render() {
        const { date, current, selected, faded, isValidDate } = this.props;
        const dayClass = classnames({
            'datetime-day': true,
        });
        const validDate = isValidDate ? isValidDate(date) : true;
        const contentClass = classnames({
            'datetime-day__content': true,
            'datetime-day__content_current': current,
            'datetime-day__content_selected': selected,
            'datetime-day__content_faded': faded,
            'datetime-day__content_disabled': !validDate,
        });
        return (
            <td
                className={dayClass}
                onClick={this.clickHandler.bind(this, validDate)}
            >
                <div className={contentClass}>
                    {date.format('D')}
                </div>
            </td>
        );
    }
}

Day.displayName = 'Day';

Day.propTypes = {
    date: propIsMoment.isRequired,
    onClick: PropTypes.func,
    isValidDate: PropTypes.func,
    current: PropTypes.bool,
    selected: PropTypes.bool,
    faded: PropTypes.bool,
};

Day.defaultProps = {
    onClick: null,
    isValidDate: null,
    current: false,
    selected: false,
    faded: false,
};

export default Day;
