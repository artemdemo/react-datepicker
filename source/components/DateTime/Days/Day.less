@import '../datetime-general';

.datetime-day {
    text-align: center;
    cursor: pointer;
}
    .datetime-day__content {
        border: 2px solid transparent;
        border-radius: 4px;
        padding: 3px;

        &:hover {
            background: @hover-color;
        }
    }
        .datetime-day__content_current {
            border-color: @hover-color;
        }
        .datetime-day__content_selected {
            background-color: @hover-color;
        }
        .datetime-day__content_faded {
            color: fade(@text-color, 35%);
        }
        .datetime-day__content_disabled {
            color: fade(@text-color, 15%);
            cursor: not-allowed;

            &:hover {
                background: transparent;
            }
        }
