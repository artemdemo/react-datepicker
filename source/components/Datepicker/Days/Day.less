@import '../datepicker-general';

.datepicker-day {
    text-align: center;
    cursor: pointer;
}
    .datepicker-day__content {
        border: 2px solid transparent;
        border-radius: 4px;
        padding: 3px;

        &:hover {
            background: @hover-color;
        }
    }
        .datepicker-day__content_current {
            border-color: @hover-color;
        }
        .datepicker-day__content_faded {
            color: fade(@text-color, 30%);
        }
