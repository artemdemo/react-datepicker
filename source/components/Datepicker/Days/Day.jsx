import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './Day.less';

class Day extends React.Component {
    clickHandler() {
        const { date, onClick } = this.props;
        onClick && onClick(date);
    }

    render() {
        const { date, current, faded } = this.props;
        const dayClass = classnames({
            'datepicker-day': true,
        });
        const contentClass = classnames({
            'datepicker-day__content': true,
            'datepicker-day__content_current': current,
            'datepicker-day__content_faded': faded,
        });
        return (
            <td
                className={dayClass}
                onClick={this.clickHandler.bind(this)}>
                <div className={contentClass}>
                    {date.format('D')}
                </div>
            </td>
        );
    }
}

Day.propTypes = {
    date: PropTypes.shape({}).isRequired,
    onClick: PropTypes.func,
    current: PropTypes.bool,
    faded: PropTypes.bool,
};

Day.defaultProps = {
    onClick: null,
    current: false,
    faded: false,
};

export default Day;
