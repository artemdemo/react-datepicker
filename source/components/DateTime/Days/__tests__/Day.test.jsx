import React from 'react';
import moment from 'moment';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Day from '../Day';

describe('<Day />', () => {
    const date = moment(1502459094000);
    const DayContainer = props => (
        <table>
            <tbody>
                <tr>
                    {props.children}
                </tr>
            </tbody>
        </table>
    );

    it('simple render', () => {
        const tree = renderer.create(
            <DayContainer>
                <Day date={date} />
            </DayContainer>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('onClick', () => {
        let clicked = false;
        const wrapper = mount(
            <DayContainer>
                <Day
                    date={date}
                    onClick={() => clicked = true}
                />
            </DayContainer>,
        );
        wrapper.find('.datetime-day').first().simulate('click');
        expect(clicked).toEqual(true);
    });

    it('isValidDate', () => {
        const tree = renderer.create(
            <DayContainer>
                <Day
                    date={date}
                    isValidDate={() => false}
                />
            </DayContainer>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('current', () => {
        const tree = renderer.create(
            <DayContainer>
                <Day
                    date={date}
                    current
                />
            </DayContainer>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('selected', () => {
        const tree = renderer.create(
            <DayContainer>
                <Day
                    date={date}
                    selected
                />
            </DayContainer>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('faded', () => {
        const tree = renderer.create(
            <DayContainer>
                <Day
                    date={date}
                    faded
                />
            </DayContainer>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
