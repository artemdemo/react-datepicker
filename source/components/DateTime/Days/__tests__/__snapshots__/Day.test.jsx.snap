// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> current 1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_current"
        >
          11
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Day /> faded 1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          11
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Day /> isValidDate 1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          11
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Day /> selected 1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_selected"
        >
          11
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Day /> simple render 1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          11
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
